docs/src/details/main-modules/images/sources.rst

```diff
@@ -0,0 +1,125 @@
+    +-----------------------+--------------+------------+
+    | Unicode string        | No           | ROM or RAM |
+    +-----------------------+--------------+------------+
+
+
+.. _image_sources_variables:
+
+Variables
+*********
+
+Images stored as variables are comprised of an :cpp:struct:`lv_image_dsc_t` struct
+plus its data.  The struct contains the following fields:
+
+:header:
+
+    :magic:       Magic number (must be LV_IMAGE_HEADER_MAGIC)
+    :cf:          Color format. See :ref:`images_color_formats`.
+    :flags:       Image flags (any of the ``LV_IMAGE_FLAGS_...`` enumeration values).
+    :w:           Width in pixels (<= 2048)
+    :h:           Height in pixels (<= 2048)
+    :stride:      Number of bytes between the beginning of one row of pixels and the beginning of the next row
+    :reserved_2:  reserved for future use
+
+:data_size:  length of ``data`` in bytes
+:data:       pointer to a byte array where the data of the image is stored (pixels
+             and sometimes indexed color palettes [for indexed color formats])
+
+These are usually stored within a project as a C file generated by one of these two
+LVGL image converters:
+
+- `LVGL Online Image Converter`_
+- Offline converter in ``./scripts/LVGLImage.py``.
+
+At this writing, the online converter only supports these formats:
+
+- RGB565
+- RGB565A8
+- RGB888
+- XRGB8888
+- ARGB8888
+
+The offline converter can convert an image into the above formats plus all the other
+supported color formats.
+
+The generated C files are compiled and linked into the resulting executable like any
+other ``const`` data.
+
+
+.. _image_sources_files:
+
+Files
+*****
+
+Files can be used as image sources by passing in the path string to the image file
+to the ``lv_<widget>_set_src()`` function as the ``src`` argument.  The string must
+begin with a printable ASCII character (range [0x20-0x7F]).  Example:
+
+.. code-block:: c
+
+    lv_image_set_src(icon, "S:my_icon.bin");
+
+
+Images stored as files are not linked into the resulting executable, and must be read
+into RAM before being drawn.  As a result, they are not as resource-friendly as
+images linked at compile time.  However, they are easier to replace without needing
+to rebuild the main program.
+
+The exception to this is BMP files which use a very low RAM footprint because the
+drawn image is taken directly from the BMP file as it is needed.  The RAM cost is
+only the size of the file buffer and the size of the area being drawn at any given
+moment.
+
+For the images module to deal with files you need to add a *Driver* for the target
+storage system to LVGL.  See the :ref:`file_system` section to learn more.
+
+
+.. _image_sources_unicode:
+
+Unicode Strings
+***************
+
+The images module Unicode strings are ``char *`` strings whose first byte contains a
+value >= 0x80.  Most often, such strings are passed as the image source like this:
+
+.. code-block:: c
+
+    lv_image_set_src(icon, LV_SYMBOL_BATTERY_FULL);
```

Comment:
It works a little bit differently, see [here](https://docs.lvgl.io/master/details/widgets/image.html#label-as-an-image).